### PR TITLE
[TT-5133] Fix the panic when analytics plugin enabled

### PR DESCRIPTION
**Pull request by furkansenharputlu**

### CODE REVIEW

File: gateway/handler_error.go [hunk 1]
```diff
@@ -335,7 +335,7 @@ func (e *ErrorHandler) HandleError(w http.ResponseWriter, r *http.Request, errMs
 			NormalisePath(&record, &e.Spec.GlobalConfig)
 		}
 
-		if e.Spec.AnalyticsPlugin.Enabled {
+		if e.Spec.AnalyticsPluginConfig != nil {
```
Comment (urbanishimwe) [inline gateway/handler_error.go:338]: can we value that field `enabled`, and also in `handler_success.go:249`
```suggestion
		if e.Spec.AnalyticsPluginConfig.Enabled && e.Spec.AnalyticsPluginConfig != nil {
```

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4118/merge
Commit: 0895b07cbe9f55a955c188818c61b2b7eb7d80be 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2473859270)

File: gateway/handler_error.go (hunk 1 above)
Comment (furkansenharputlu) [reply gateway/handler_error.go:338]: but it is just not `nil` when it is enabled.

Comment (urbanishimwe) [reply gateway/handler_error.go:338]: it may not be nil, and still be disabled... like the case for `customDomain`

**Review by urbanishimwe — APPROVED** [review]

**Review by jeffy-mathew — APPROVED** [review]

File: gateway/handler_error.go (hunk 1 above)
Comment (furkansenharputlu) [reply gateway/handler_error.go]: I applied another solution, this is not an issue anymore.

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4118/merge
Commit: e27f25a94e6026df074e0eff3dc34669367540b5 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2473980498)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 5f051b7b1327cfcdccfcf0651504f46d8b5d1e24 Fix the panic when analytics plugin enabled (#4118)
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2474040317)

**Pull request merged**